### PR TITLE
Improves event handling

### PR DESCRIPTION
**Pull request by mecha**
Incorporates a solution that is related to the revision of issue #4. See my first comment on that issue.

----

Adds awareness of an event factory. The state machine now uses a factory for creating events. It also ensures that the static event params do not override the params determined by the state machine, by swapping the merging order.

### CODE REVIEW

File: src/EventStateMachine.php [hunk 1]
```diff
@@ -329,6 +350,30 @@ protected function _setEventManager($eventManager)
...
     }
 
+    /**
+     * Retrieves the event factory associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return EventFactoryInterface The event factory.
+     */
+    protected function _getEventFactory()
```
Comment (XedinUnknown) [inline src/EventStateMachine.php:360]: Would you say that traits from [here](https://github.com/RebelCode/modular/tree/develop/src/Events) would be helpful in this case?

**Review by XedinUnknown — APPROVED** [review]

File: src/EventStateMachine.php (hunk 1 above)
Comment (mecha) [reply src/EventStateMachine.php:360]: Perhaps. To replace the getters/setters, maybe. But for something like the event factory, I would just not have getters and setters to begin with.

Comment (XedinUnknown) [reply src/EventStateMachine.php:360]: You would not have getters and setters for the factory? Why not?

Comment (mecha) [reply src/EventStateMachine.php:360]: Single type, interface, non-null. It's unnecessary and more to test.

Comment (XedinUnknown) [reply src/EventStateMachine.php:360]: Copy-pasting the getters and setters is what's more to test, IMHO.

Comment (mecha) [reply src/EventStateMachine.php:360]: Who said anything about copy-paste getters and setters? I literally just said I would not have getters and setters.

Comment (XedinUnknown) [reply src/EventStateMachine.php:360]: I see. Anyhow, extracting the traits could be helpful; see [card](https://github.com/orgs/RebelCode/projects/2#card-11665009).

**Pull request merged**